packages/workspace/src/core/project-graph/project-graph-builder.ts

```diff
@@ -21,12 +21,16 @@ export class ProjectGraphBuilder {
     }
   }
 
-  addNode(node: ProjectGraphNode) {
+  addNode(node: ProjectGraphNode, mergeData: (d: any) => any = x => x) {
+    const prev = this.nodes[node.name];
     this.nodes[node.name] = node;
+    if (prev) {
+      node.data = mergeData(prev.data);
+    }
   }
 
   addDependency(
-    type: DependencyType,
+    type: DependencyType | string,
```

Comment:
We should keep this to just `DependencyType` right?

Reply:
No, I think we may want to remove `DependencyType` completely since it could be extended with any string (if/when we allow plugins). For now It's unioned because it could still be useful to keep core types?